weldx/visualization.py

```diff
@@ -83,3 +181,575 @@ def set_axes_equal(axes):
+               LocalCoordinateSystem
+            The time steps that should be plotted
+        time_ref : pandas.Timestamp
+            A reference timestamp that can be provided if the ``time`` parameter is a
+            `pandas.TimedeltaIndex`
+
+        """
+        lcs = self._lcs.interp_time(time, time_ref)
+        coordinates, orientation = self._get_coordinates_and_orientation(lcs)
+
+        self._update_positions(coordinates, orientation)
+
+    def update_time_index(self, index):
+        """Update the plotted time step.
+
+        Parameters
+        ----------
+        index : int
+            The array index of the time step
+
+        """
+        coordinates, orientation = self._get_coordinates_and_orientation(
+            self._lcs, index
+        )
+        self._update_positions(coordinates, orientation)
+
+
+class CoordinateSystemManagerVisualizerK3D:
+    """Visualizes a `weldx.CoordinateSystemManager` using k3d."""
+
+    color_table = [0xFF0000, 0x00AA00, 0x0000FF, 0xAAAA00, 0xFF00FF, 0x00FFFF]
+
+    def __init__(
+        self,
+        csm,
+        coordinate_systems=None,
+        colors=None,
+        reference_system=None,
+        title=None,
+        limits=None,
+        time=None,
+        time_ref=None,
+        show_labels=True,
+        show_origins=True,
+        show_traces=True,
+        show_vectors=True,
+    ):
+        """Create a `CoordinateSystemManagerVisualizerK3D`.
+
+        Parameters
+        ----------
+        csm : weldx.CoordinateSystemManager
+            The `CoordinateSystemManager` that should be visualized
+        time : pandas.DatetimeIndex, pandas.TimedeltaIndex, List[pandas.Timestamp], or \
+               LocalCoordinateSystem
+            The time steps that should be plotted
+        time_ref : pandas.Timestamp
+            A reference timestamp that can be provided if the ``time`` parameter is a
+            `pandas.TimedeltaIndex`
+
+        """
+        if time is None:
+            time = csm.time_union()
+        if time is not None:
+            csm = csm.interp_time(time=time, time_ref=time_ref)
+        self._csm = csm.interp_time(time=time, time_ref=time_ref)
+        self._current_time_index = 0
+
+        if coordinate_systems is None:
+            coordinate_systems = csm.coordinate_system_names
+        if reference_system is None:
+            reference_system = self._csm._root_system_name
+        if limits is None:
+            limits = [-1, 1]
+
+        # create controls
+        self._controls = self._create_controls(
+            time,
+            reference_system,
+            show_labels,
+            show_origins,
+            show_traces,
+            show_vectors,
+        )
+
+        # create plot using dict comprehension
+        self._color_generator = color_generator_function()
+        plot = k3d.plot(
+            grid_auto_fit=False,
+            camera_auto_fit=False,
```

Comment:
Why not `True`? It difficult to position the camera for welding examples this way

Reply:
Because I never deleted those lines 😆  I will remove them. Additionally, I will provide the option to access the plot object directly so that one can adjust it as needed.